### PR TITLE
Hot attach suite

### PR DESCRIPTION
**Pull request by koln67**
Hot attach test suite

Included some edits to add sequential write and read tests

### CODE REVIEW

File: imagetest/test_suites/hotattach/hot_attach_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	projectZoneSlice := strings.Split(string(projectZoneUrl), "/")
+	if strings.ToLower(projectZoneSlice[0]) != "projects" || strings.ToLower(projectZoneSlice[2]) != "zones" || len(projectZoneSlice) != 4 {
+		return "", "", fmt.Errorf("returned string for vm metata was the wrong format: got %s", projectZoneUrl)
+	}
+
+	// return format is (projectNumber, instanceZone, nil)
+	return projectZoneSlice[1], projectZoneSlice[3], nil
+}
+
+// TestFileHotAttach is a test which checks that a file on a disk is usable, even after the disk was detached and reattached.
```
Comment (a-crate) [inline imagetest/test_suites/hotattach/hot_attach_test.go:43]: The comment here says that this test checks that a file on disk is usable after reattach but this test doesn't look like it's reading or writing any files, just calling the detach & attach APIs.

Comment (koln67) [reply imagetest/test_suites/hotattach/hot_attach_test.go:87]: added

**Comment by a-crate** [comment]
This include all the changes from [throughput testing](https://github.com/GoogleCloudPlatform/guest-test-infra/pull/781) too, do they both need review? LGTM but neither has a hold and I don't want to create merge conflicts by lgtm-ing before it's ready.

**Comment by koln67** [comment]
Both PRs should be ready, though I might make minor changes to hotattach. 

It might be easiest to merge the other througput PR, and then merge shapevalidation. That would ensure no merge conflicts, and I would create a new PR with the hotattach tests.

**Pull request closed without merging**